src/gen_rules.ml

```diff
@@ -28,30 +28,53 @@ module Gen(P : Params) = struct
      +-----------------------------------------------------------------+ *)
 
   module Eval_modules = Ordered_set_lang.Make(struct
-      type t = Module.t
-      let name = Module.name
+      type t = (Module.t, string) result
```

Comment:
actually i think it's important to use `(Module.t, string * Loc.t)` otherwise the reported locations won't be the same when you start adding a `modules_without_implementations` field for instance